crates/onnx-ir/src/proto_conversion.rs

```diff
@@ -242,21 +242,27 @@ impl TryFrom<ValueInfoProto> for Argument {
             }
         };
 
-        let tensor_type = TensorType {
-            dim: tensor_proto.shape.dim.len(),
-            elem_type,
-            shape: Some(
-                tensor_proto
-                    .shape
-                    .dim
-                    .iter()
-                    .map(|x| x.dim_value() as Dim)
-                    .collect(),
-            ),
+        let ty = if tensor_proto.shape.dim.is_empty() {
```

Comment:
I could be wrong, but aren't scalars sometimes also represented as a tensor with `dim: 0`? 🤔

Reply:
In ONNX? Yes, that's what this is checking for. In Burn, I think a Tensor with dim = 0 is unwanted.

Note that `tensor_proto.shape.dim` is a `Vec` of the shape dimensions (maybe another rename candidate, similar to `Tensor::dim` -> `rank` ), so `dim = tensor_proto.shape.dim.len()` (see also line 251 below), therefore the check  `dim == 0` becomes `tensor_proto.shape.dim.len() == 0`, and `clippy` ahs hammered into me to replace `Vec::len() == 0` with `Vec::is_empty()`